readthedocs/urls.py

```diff
@@ -13,12 +13,14 @@
     HomepageView,
     SupportView,
     do_not_track,
+    server_error_404,
     server_error_500,
 )
 from readthedocs.search.views import GlobalSearchView
 
 admin.autodiscover()
 
+handler404 = server_error_404
```

Comment:
I have actually been confused about what to do wrt `ProjectDetailView` and `project.urls.public`

https://github.com/readthedocs/readthedocs.org/blob/7128a001f900b0236d441053543bf324c2c452af/readthedocs/projects/urls/public.py#L32-L36

I haven't read the implementation or pondered anything here, thinking that it has very little or no impact to add a nice 404 here.

The intention was that we can raise the exceptions from anywhere in the codebase. I also wondered about moving them out of proxito, but they are inspired by proxito internals, so I think they belong there :)

Reply:
Curious if you have thoughts on this?

Reply:
I'm +1 on having nicer 404 pages on the main application, but if we do so, we probably want its own implementation separate from the proxito one, since proxito has restrictions that the main app doesn't, specially: URLs from the dashboard aren't present in proxito, we have access to the current user session in 404 from the main app. So I think it makes sense to have them separate.

Reply:
Great! It sounds like this modular scoping will be much more clarified when moving 404 templates from the main app to proxito, as you already suggested!

Once all "proxito-scoped" code and template logic lives in proxito, the main application should be able to use or derive from it as is feasible?